web_responsive/static/src/js/web_responsive.js

```diff
@@ -292,10 +295,39 @@ odoo.define('web_responsive', function(require) {
         new AppDrawer();
     });
 
+    // if we are in small screen change default view to kanban if exists
+    ViewManager.include({
+        get_default_view: function() {
+            var default_view = this._super()
+            if (config.device.size_class <= config.device.SIZES.XS &&
+                default_view != 'kanban' &&
+                this.views['kanban']){
+                    default_view = 'kanban';
+            };
+            return default_view;
+        },
+    });
+
+    // if we are in small screen change One2many field view to kanban if exists
+    FieldOne2Many.include({
```

Comment:
I think @pedrobaeza means that with the above code in `ViewManager` should work. Try to remove this chunk and see if it works. Otherwise, it's perfectly OK to have this here.

I'd thank you a lot if you can document this behavior in the addon's README, since it's somewhat hidden and unexpected otherwise.